src/sentry/api/event_search.py

```diff
@@ -1250,25 +1272,117 @@ def get_filter(query=None, params=None):
     return eventstore.Filter(**kwargs)
 
 
+class PseudoField(object):
+    def __init__(self, name, alias, expression=None, expression_fn=None, result_type=None):
+        self.name = name
+        self.alias = alias
+        self.expression = expression
+        self.expression_fn = expression_fn
+        self.result_type = result_type
+
+    def get_expression(self, params):
+        if isinstance(self.expression, (list, tuple)):
+            expression = deepcopy(self.expression)
+            return expression
+        elif self.expression_fn is not None:
+            expression = self.expression_fn(params)
+            return expression
+        return None
+
+    def get_field(self, params=None):
+        expression = self.get_expression(params)
+        if expression is not None:
+            expression.append(self.alias)
+            return expression
+        return self.alias
+
+    def validate(self):
+        assert self.alias is not None, u"{}: alias is required".format(self.name)
+        no_expression = self.expression is None
+        no_expression_fn = self.expression_fn is None
+        assert (
+            no_expression != no_expression_fn
+        ), u"{}: only one of expression, expression_fn is allowed".format(self.name)
+
+
+# This function will only ever be called with 1 set of parameters in the lifetime of a
+# single request. So cache its results to avoid doing more lookups than necessary.
+@functools.lru_cache(maxsize=1)
+def key_transaction_expression(user_id, organization_id, project_ids):
+    if user_id is None or organization_id is None or project_ids is None:
+        raise InvalidSearchQuery("Missing necessary meta for key transaction field.")
+
+    key_transactions = (
+        KeyTransaction.objects.filter(
+            owner__id=user_id, organization__id=organization_id, project__id__in=project_ids,
```

Comment:
```suggestion
            owner_id=user_id, organization_id=organization_id, project_id__in=project_ids,
```

Using the `__` will likely do joins as you're setting conditions on the associations. Using column names should avoid that.